java/src/org/openqa/selenium/grid/security/SecretOptions.java

```diff
@@ -41,7 +41,8 @@ public SecretOptions(Config config) {
 
   public Secret getRegistrationSecret() {
     String secret = "";
-    if ((isSecure()) && !config.get(SERVER_SECTION, "registration-secret").isPresent()) {
```

Comment:
Why is this one needed?

Reply:
Ah, I saw cross servers, there are 2 conditions with OR. For example, user enabled isSelfSigned(), it should behave the same.
Since the method defined but not called, so I guess it could be missed.

Reply:
Should be fine, right.